Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -678,30 +688,32 @@ void DirectoryBlobReader::SetPartitions(std::vector<PartitionWithType>&& partiti
   }
   m_data_size = partition_address;
 
-  m_nonpartition_contents.AddReference(PARTITION_TABLE_ADDRESS, m_partition_table);
+  m_nonpartition_contents.Add(PARTITION_TABLE_ADDRESS, std::move(partition_table));
 }
 
 // This function sets the header that's shortly before the start of the encrypted
 // area, not the header that's right at the beginning of the encrypted area
 void DirectoryBlobReader::SetPartitionHeader(DirectoryBlobPartition* partition,
-                                             u64 partition_address)
+                                             u64 partition_address,
+                                             std::unique_ptr<DiscIO::VolumeDisc> volume)
 {
   constexpr u32 TMD_OFFSET = 0x2c0;
   constexpr u32 H3_OFFSET = 0x4000;
 
   const std::optional<DiscIO::Partition>& wrapped_partition = partition->GetWrappedPartition();
+  // These should either both be set or both be null, but better to be safe.
+  bool wrapped = wrapped_partition && volume;
   const std::string& partition_root = partition->GetRootDirectory();
 
   u64 ticket_size;
-  if (wrapped_partition)
+  if (wrapped)
   {
-    const auto& ticket = m_wrapped_volume->GetTicket(*wrapped_partition).GetBytes();
-    auto& new_ticket = m_extra_data.emplace_back(ticket);
+    std::vector<u8> new_ticket = std::move(volume->GetTicket(*wrapped_partition).GetBytes());
```

Comment:
Unnecessary `std::move`. (You already have an rvalue.)

```suggestion
    std::vector<u8> new_ticket = volume->GetTicket(*wrapped_partition).GetBytes();
```